kernel/os/syscfg.yml

```diff
@@ -190,6 +195,9 @@ syscfg.defs:
             If set, run time is measured in cpu time ticks rather than OS time
             ticks.
         value: 0
+    MEMFAULT_ENABLE:
+        description: 'Enables Memfault SDK features'
```

Comment:
This should be moved to `util/memfault_sdk/syscfg.yml`

Reply:
util/memfault_sdk is currently only included inside `kernel/os/pkg.yml` if `MEMFAULT_ENABLE` is set. Putting `MEMFAULT_ENABLE` inside `util/memfault_sdk/syscfg.yml` would create a catch-22 circular dependency

Reply:
We do not need the dependency there

Reply:
Right, os no longer needs to pull in the package, I can move it to the app side. Changed